File: Backend/farmguider/src/main/java/pl/edu/uwm/farmguider/models/cow/dtos/CowResponseDTO.java
package pl.edu.uwm.farmguider.models.cow.dtos;

import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

public record CowResponseDTO(Long cowId,
                             String cowName,
                             String earTagNumber,
                             LocalDate dateOfBirth,
                             String gender,
                             BigDecimal currentWeight,
                             BigDecimal latestMilkingQuantity,
                             LocalDateTime latestMilkingDate,
                             LocalDate latestWeightMeasurementDate) {

    @Builder
    public CowResponseDTO {}

}

File: Frontend/farmguider-app/src/entities/CowResponseDTO.ts
export default interface CowResponseDTO {
    cowId: number;
    cowName: string | null;
    earTagNumber: string;
    dateOfBirth: Date;
    gender: string;
    currentWeight: number;
    latestMilkingQuantity: number;
    latestMilkingDate: Date | null;
    latestWeightMeasurementDate: Date | null;
}